### ISSUE TITLE
Divide by zero when calling HMM_Slerp with two identity quaternions

### ISSUE THREAD

**Issue by shinymerlyn** [HandmadeMath/HandmadeMath#135]
If the angle between two quaternions is zero (or very close to zero), you end up with a divide by zero/infinity.

The way GLM solves this is detecting if `Cos_Theta` is very near 1.0f (greater than `1.0f - epsilon`), and if so it reverts to a lerp.

Repro:
```
hmm_quaternion result = HMM_Slerp(
    (hmm_quaternion) { .X = 0.0f, .Y = 0.0f, .Z = 0.0f, .W = 1.0f },
    0.5f,
    (hmm_quaternion) { .X = 0.0f, .Y = 0.0f, .Z = 0.0f, .W = 1.0f });
```

**Comment by shinymerlyn** [comment]
https://github.com/g-truc/glm/blob/master/glm/ext/quaternion_common.inl#L41

**Comment by bvisness** [comment]
Fixed in HMM 2.0.
